language/stdlib/modules/account_tracking.move

```diff
@@ -0,0 +1,173 @@
+///////////////////////////////////////////////////////////////////////////
+// Every account on chain has a default LimitsDefinition this is stored
+// under a specified address. However, certain
+// accounts (e.g. VASPs) may have specific limits definitions. In order to
+// set these specifi account limits, the account publishes a proposed
+// LimitsDefinition resource under their root authority account (in the
+// case of a VASP, under the root account). The association then certifies
+// this LimitsDefinition resource. After that, it will be used for tracking
+// account inflow/outflow/holding.
+//
+// With this information, when we verify that an account A can send or
+// receive money there are two cases the consider:
+// 1. This account is a "normal" (unhosted) account: in this case the
+//    account limits window is held under A and the default limits definition
+//    is used.
+// 2. The account A is part of a multi-account on-chain entity (e.g. VASP).
+//    In this case we trace back to the root authority for the account and
+//    use the limit definition under the root account, and the window under
+//    the root account.
+///////////////////////////////////////////////////////////////////////////
+
+// Error codes:
+// 3000 -> INVALID_INITIALIZATION_SENDER
+// 3001 -> UNKNOWN_ACCOUNT_TYPE
+// 3002 -> ATTEMPTED_OPERATION_ON_EMPTY_ACCOUNT_TYPE
+
+address 0x0:
+
+module AccountTrack {
+    use 0x0::Transaction;
+    use 0x0::VASP;
+    use 0x0::Unhosted;
+    use 0x0::Empty;
+    use 0x0::AccountType;
+    use 0x0::AccountLimits;
+
+    // An `AccountLimitsCapability` holds the capabilities needed in order
+    // to call other privileged functions in other modules.
+    resource struct AccountLimitsCapability {
+        // Allows functions in this module to mutate
+        // `AccountLimits::Window` values.
+        account_limits_cap: AccountLimits::MutationCapability,
+        // Allows functions in this module to mutate the interior state of
+        // account types
+        update_cap: AccountType::UpdateCapability,
+    }
+
+    resource struct CallingCapability { }
```

Comment:
A good question! I'll add a comment.

The key thing is to note that the operations `update_deposit_limits` and `update_withdrawal_limits` in this module are effectful operations over account information at addresses that may be different from the sender. We therefore need to restrict who can call these so that not just anyone can update e.g. the amount of money withdrawn from an account. This capability is what is used to enforce the operations restriction.